### PR TITLE
feat: IdentifyPrimaryAccountStage (#164)

### LINKED ISSUE #164: Pipeline stage: Identify primary account
## Description

Implement the `IdentifyPrimaryAccountStage` — runs on first sync only to detect the user's primary bank account by finding consistent income deposits and outbound transfers to other synced accounts.

**Parent Epic:** #160
**Depends on:** #162 (Orchestrator)

## Algorithm

**File:** `app/Services/PipelineStages/IdentifyPrimaryAccountStage.php`

- **key:** `'identify-primary-account'`
- **shouldRun:** `$context->isFirstSync && $context->user->primary_account_id === null`

### Detection Logic

1. Load all active accounts for the user (Transaction/Savings type via `AccountClass`)
2. For each account, find **Credit** transactions grouped by `clean_description ?? description`
3. Filter to groups with **2+ occurrences**
4. For each group, score by:
   - **Amount consistency** — low standard deviation of amounts
   - **Interval consistency** — low standard deviation of days between consecutive transactions
   - **Occurrence count** — more = higher confidence
   - **Salary-size signal** — amount > $500 (50000 cents)
5. Pick highest-scoring candidate across all accounts
6. **Secondary signal:** count outbound transfers (debits with `transfer_pair_id` pointing to other synced accounts) from this account — adds to confidence
7. Create `AnalysisSuggestion` (type: `PrimaryAccount`)
8. Set context fields for Stage 2: `primaryAccountId`, `incomeAmount`, `incomeFrequency`, `incomeDescription`, `incomeTransactionIds`

### Frequency Detection

Given sorted transaction dates for recurring credit:
- Calculate all intervals (days between consecutive occurrences)
- Compute median interval
- Map: 5-9d → Weekly, 12-16d → Fortnightly, 27-35d → Monthly
- Reject if coefficient of variation (std dev / median) > 0.3

### Suggestion Payload Schema

```json
{
  "account_id": 42,
  "account_name": "Everyday Account",
  "income_amount": 300000,
  "income_frequency": "fortnightly",
  "income_description": "SALARY DEPOSIT",
  "confidence_score": 0.92,
  "matched_transaction_ids": [101, 215, 330],
  "outbound_transfer_count": 5
}
```

## Tasks

- [ ] Create `IdentifyPrimaryAccountStage` implementing `PipelineStageContract`
- [ ] Implement income pattern detection (grouping, scoring)
- [ ] Implement frequency inference from interval analysis
- [ ] Implement outbound transfer signal detection
- [ ] Implement confidence scoring
- [ ] Set context fields for downstream Stage 2
- [ ] Write tests:
  - User with clear primary account (consistent salary)
  - User with multiple income sources (picks strongest)
  - User with no clear income pattern (stage produces no suggestion)
  - Transfers from primary to savings/credit card accounts boost confidence
  - Stage skips if not first sync or primary already set
- [ ] Run `op ci` to verify

## Acceptance Criteria

- Correctly identifies the account with consistent income deposits
- Frequency detection handles weekly, fortnightly, monthly patterns
- Outbound transfers serve as secondary confirmation signal
- Sets context fields so Stage 2 can use the results
- Gracefully handles edge cases (no income, multiple incomes, irregular patterns)
- Only runs on first sync when primary_account_id is null
- All tests pass, `op ci` green

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

- Implements `IdentifyPrimaryAccountStage` — the second pipeline stage that detects the user's primary bank account on first sync by finding consistent income deposits (salary patterns) across Transaction and Savings accounts
- Uses a 4-component confidence scoring system (interval regularity, amount consistency, occurrence count, salary-size threshold) with an outbound transfer bonus
- Maps income intervals to `PayFrequency` (Weekly/Fortnightly/Monthly) for downstream use by the future `IdentifyPayCycleStage` (#165)
- Only executes on first sync when no primary account is set (`shouldRun` guard), adding zero cost to subsequent pipeline runs

## Changes

- **Created** `app/Services/PipelineStages/IdentifyPrimaryAccountStage.php` — full stage implementation
- **Created** `tests/Feature/Services/PipelineStages/IdentifyPrimaryAccountStageTest.php` — 23 tests (49 assertions)
- **Modified** `app/Providers/AppServiceProvider.php` — registered stage first in pipeline
- **Modified** `database/factories/AnalysisSuggestionFactory.php` — added `primaryAccount()` state

## Test plan

- [x] `op test.filter IdentifyPrimaryAccountStage` — 23 tests pass
- [x] `op test.filter TransactionAnalysisPipelineTest` — 19 existing pipeline tests pass
- [x] `op lint.dirty` — Pint clean
- [x] `op ci` — full CI green (1112 tests, PHPStan clean)

Closes #164

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/Services/PipelineStages/IdentifyPrimaryAccountStage.php [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+            ->whereNull('transfer_pair_id')
+            ->current()
+            ->orderBy('post_date')
+            ->get();
+
+        if ($credits->count() < self::MIN_OCCURRENCES) {
+            return null;
+        }
+
+        $grouped = $credits->groupBy(fn (Transaction $t) => $this->normalizeDescription($t->description));
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/PipelineStages/IdentifyPrimaryAccountStage.php]: Fixed — `normalizeDescription()` now accepts a `Transaction` and uses the `merchant_name ?? clean_description ?? description` fallback chain, matching `IdentifyRecurringTransactionsStage`. Test helper updated to null out `merchant_name` and `clean_description` so tests control grouping via the explicit `description` field.

File: app/Services/PipelineStages/IdentifyPrimaryAccountStage.php [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            + (self::WEIGHT_SALARY * $salaryScore);
+
+        return [
+            'account' => $account,
+            'confidence' => round($confidence, 4),
+            'median_amount' => (int) $medianAmount,
+            'frequency' => $frequency,
+            'description' => $description,
+            'transaction_ids' => $sorted->pluck('id')->all(),
+        ];
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/PipelineStages/IdentifyPrimaryAccountStage.php:218]: Fixed — changed `(int) $medianAmount` to `(int) round($medianAmount)` to avoid systematic floor bias on even-sized groups.

File: app/Services/PipelineStages/IdentifyPrimaryAccountStage.php [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+                'outbound_transfer_count' => $transferCount,
+            ],
+        ]);
+
+        return new StageResult(
+            success: true,
+            stage: self::STAGE_KEY,
+            suggestionIds: [$suggestion->id],
+        );
+    }
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Services/PipelineStages/IdentifyPrimaryAccountStage.php:127]: By design — `PipelineContext` is intentionally readonly. The pipeline uses the Blackboard pattern: stages communicate through persisted `AnalysisSuggestion` records. The future `IdentifyPayCycleStage` (#165) will query for the `PrimaryAccount` suggestion by `pipeline_run_id` and `SuggestionType`. No changes to context/result types needed.
